libs/island-ui/core/src/lib/Typography/Typography.tsx

```diff
@@ -35,28 +36,37 @@ export interface TypographyProps {
   paddingTop?: ResponsiveSpace
   paddingBottom?: ResponsiveSpace
   paddingY?: ResponsiveSpace
+  marginTop?: ResponsiveSpace
+  marginBottom?: ResponsiveSpace
+  marginY?: ResponsiveSpace
   fontWeight?: keyof typeof fontWeightStyles
 }
 
 export const Typography = ({
   id,
   variant,
-  as: Cmp = 'p',
   children,
   color,
   truncate,
   links,
   paddingTop,
   paddingBottom,
   paddingY,
+  marginTop,
+  marginBottom,
+  marginY,
   fontWeight,
+  as = 'p',
 }: TypographyProps) => {
   const resolvedPaddingTop = paddingTop ?? paddingY
   const resolvedPaddingBottom = paddingBottom ?? paddingY
+  const resolvedMarginTop = marginTop ?? marginY
+  const resolvedMarginBottom = marginBottom ?? marginY
 
   return (
-    <Cmp
+    <Box
       id={id}
+      component={as}
```

Comment:
couldn't we just make Box resolve the spacing styles and remove them from typography.treat?
```suggestion
    <Box
      id={id}
      component={as}
      marginTop={marginTop}
      marginBottom={marginBottom}
      marginY={marginY}
      paddingTop={paddingTop}
      paddingBottom={paddingBottom}
      paddingY={paddingY}
```